recipes/sge_install.rb

```diff
@@ -29,20 +29,30 @@
     not_if { ::File.exist?(sge_tarball) }
   end
 
+  # Patch file for SGE sources
+  cookbook_file 'sge_source.patch' do
+    path '/tmp/sge_source.patch'
+    user 'root'
+    group 'root'
+    mode '0644'
+  end
+
   # Install SGE
   bash 'make install' do
     user 'root'
     group 'root'
     cwd Chef::Config[:file_cache_path]
     environment 'SGE_ROOT' => '/opt/sge'
     code <<-SGE
+      set -e
       tar xf #{sge_tarball}
-      cd sge-#{node['cfncluster']['sge']['version']}/source
+      cd sge-#{node['cfncluster']['sge']['version']}
+      patch -p1 < /tmp/sge_source.patch
```

Comment:
I would like to apply the patch only for ubuntu18, same logic we do for torque. Let discuss about it

Reply:
The idea is to get rid of "ifs" by leaving a unique flow and managing the different behaviors, if needed, in the specific file. By default the patch is empy, it contains data only for ubuntu 18

Reply:
In this way, if a new OS will require a different patch we could just add the new file without having to change the main code

Reply:
I don't like this approach for a couple of reasons:
* the patch for ubuntu18 is applied also for the other OSs which do not require it
* if a new OS will require a different patch, as you said, it could be possible that is not compatible with ubuntu18 or the other OSs
* in the long run I would like it solved upstream, so no patch would be required

Reply:
- As I said, to the other OSs is applied an empty patch (the default one)
- If a new OS will require a different patch we will just need to put this different patch in the file files/<new OS>/sge_source.patch

Maybe it's helpful to remind that the patch is managed as a cookbook file, so each OS will have it's own patch (and just that one)

Reply:
ok good, I didn't see the patch is located in files/ubuntu-18.04/sge_source.patch